pandas/io/formats/style.py

```diff
@@ -151,7 +151,8 @@ def __init__(
 
         def default_display_func(x):
             if is_float(x):
-                return "{:>.{precision}g}".format(x, precision=self.precision)
+                display_format = "{0:.{precision}f}".format(x, precision=self.precision)
+                return display_format.rstrip("0").rstrip(".")
```

Comment:
what is the reason for this?

why is the output not consistent with the standard terminal output?

```python
>>>
>>> import pandas as pd
>>>
>>> pd.__version__
'0.25.0+227.g347ad8564'
>>>
>>> df = pd.DataFrame(data=[[1.0, 2.0090], [3.2121, 4.566]], columns=["a", "b"])
>>>
>>> with pd.option_context("precision", 1):
...     print(df)
...
     a    b
0  1.0  2.0
1  3.2  4.6
>>> with pd.option_context("precision", 2):
...     print(df)
...
      a     b
0  1.00  2.01
1  3.21  4.57
>>> with pd.option_context("precision", 3):
...     print(df)
...
       a      b
0  1.000  2.009
1  3.212  4.566
>>>

```

Reply:
my understanding of https://github.com/pandas-dev/pandas/pull/13897#issuecomment-237551157 is that ints do not have decimal points and trailing zeros but that floats should follow the precision setting @TomAugspurger ?

The comments also states that "Ideally df.style would have the same output as df._repr_html() by default."

![image](https://user-images.githubusercontent.com/13159005/63769983-0daf8100-c8cc-11e9-9db9-7186bdae40a9.png)

Reply:
@simonjayhawkins  
The modification to make the program works as the way u said above is fair simple ( only remove rstrip ).
I really wanna just have the verdict of @TomAugspurger to make some last changes to the pull request.

Reply:
I'm not sure what's best here, but matching the output of the main DataFrame repr is probably best.

Reply:
> I'm not sure what's best here, but matching the output of the main DataFrame repr is probably best.

Ok will match it now!

Reply:
Just changed it. 
Now its working the same way as DataFrame repr is.
@TomAugspurger 

<img width="540" alt="Captura de Tela 2019-09-03 às 08 51 30" src="https://user-images.githubusercontent.com/26484794/64171088-2669ea80-ce28-11e9-8a63-01f1e0116d40.png">